azure-client-runtime/src/main/java/com/microsoft/azure/AzureClient.java

```diff
@@ -400,7 +400,8 @@ public Boolean call(PollingState<T> tPollingState) {
         pollingState.withResourceType(resourceType);
         pollingState.withSerializerAdapter(restClient().serializerAdapter());
         if (pollingState.isStatusTerminal()) {
-            if (pollingState.isStatusSucceeded()
+            if (pollingState.statusCode() != 204 
```

Comment:
LGTM, @jianghaolu improved it by adding one more check so that we don't make a call if the resource type is Void.